transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/ButtonResponse.kt

```diff
@@ -9,9 +9,32 @@ import kotlinx.serialization.Serializable
  * @param payload the payload of the quick reply.
  * @param type the type of quick reply.
  */
+
 @Serializable
 data class ButtonResponse(
     val text: String,
     val payload: String,
-    val type: String,
-)
+) : Action(type = Type.Button)
+
+@Serializable
+data class LinkAction(
+    val text: String,
+    val url: String,
+) : Action(type = Type.Link)
+
+@Serializable
+data class PostbackAction(
+    val text: String,
+    val payload: String,
+) : Action(type = Type.Postback)
+
+@Serializable
+open class Action(
```

Comment:
consider using a sealed class for Action inheritance. 
This should provide a more strict approach for inheritance disallowing customers accidentally extend Action class. 
It is also more clean when used in `when` statements.